gcp/volume.go

```diff
@@ -184,6 +195,45 @@ func (c *Client) getVolumeByRegion(region string) ([]volumeResult, error) {
 	return volumes, nil
 }
 
+func (c *Client) findRegionbyID(VolumeID string) (volumeResult, error) {
```

Comment:
should be  findVolumeByIDAcrossRegions  
how expensive an operation is it if we don't give a region?

Reply:
To stay API compatible, this extends the current getVolumeByID. In getVolumeByID it calls findRegionByID, if region is not set (which is/was a requirement for getVolumeByID to work). findRegionByID does one additional API call (enumerate all volumes) and filters the list by ID to extract the region. This region information is then used by getVolumeByID to do its job. tl;dr: If no region is provided, we find the missing region by filtering all volumes for given UUID.
Since this extends getVolumeByID and is not substituting it, I don't think findVolumeByIDAcrossRegions would be an appropriate name. Open to discuss.

Reply:
this function returns a volume (return resultVolume, nil) not a region.

Reply:
Yes, you are correct. I will change the function name.